File: benchmarks/xlang/src/collatz.R
steps <- function(n) {
  c <- 0L
  while (n != 1) {
    if (n %% 2 == 0) n <- n %/% 2 else n <- 3 * n + 1
    c <- c + 1L
  }
  c
}
t0 <- proc.time()[["elapsed"]]
total <- 0
for (i in 1:300000) total <- total + steps(i)
cat("CHECK", format(total, scientific = FALSE), "\n")
cat("MS", as.integer((proc.time()[["elapsed"]] - t0) * 1000), "\n")

File: benchmarks/xlang/src/collatz.cpp
#include <cstdio>
#include <chrono>
long steps(long n) {
    long c = 0;
    while (n != 1) { n = n % 2 == 0 ? n / 2 : 3 * n + 1; c++; }
    return c;
}
int main() {
    auto t0 = std::chrono::steady_clock::now();
    long total = 0;
    for (long i = 1; i <= 300000; i++) total += steps(i);
    auto ms = std::chrono::duration_cast<std::chrono::milliseconds>(
        std::chrono::steady_clock::now() - t0).count();
    printf("CHECK %ld\nMS %lld\n", total, (long long)ms);
}
